### ISSUE TITLE
[fixed - see new image] Can't log in after running

### ISSUE THREAD

**Issue by the4tress** [harvard-itsecurity/docker-misp#1]
I built the container, then initiated the database:

```
docker run -it --rm \
    -v /misp-db:/var/lib/mysql \
    harvarditsecurity/misp /init-db
```

Which gave me the following output:

```
chown: changing ownership of '/var/lib/mysql': Operation not permitted
Cannot change ownership of the database directories to the 'mysql'
user.  Check that you have the necessary permissions and try again.
161014 05:14:41 mysqld_safe Logging to syslog.
161014 05:14:41 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
161014 05:14:42 mysqld_safe mysqld from pid file /var/run/mysqld/mysqld.pid ended
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
```

I ran the docker command again and it seemed to complete. Then I ran the next command:

```
docker run -it -d \
    --name misp \
    -p 443:443 \
    -p 80:80 \
    -p 3306:3306 \
    -v /misp-db:/var/lib/mysql \
    harvarditsecurity/misp
```

That seemed to work as well. So then I tried to log in. The initial page loaded, but when I try to log in I get the following error:

![asdf](https://cdn.pbrd.co/images/eJQY7M0YO.png)

Any thoughts?

**Comment by routis** [comment]
I have also the same problem

**Comment by robinlennox** [comment]
Fix: It's a permissions issue.

To get it working run the following commands:

`rm -rf /misp-db`
`docker run -it --rm     -v /misp-db:/var/lib/mysql harvarditsecurity/misp /bin/bash -c "/etc/rc.local; chown -R mysql:mysql /var/lib/mysql"`
`docker run -it --rm     -v /misp-db:/var/lib/mysql harvarditsecurity/misp /init-db`

**Comment by ventz** [comment]
@the4tress @routis @robinlennox - sorry for the late response, for some reason these never generated an email and I am just seeing this issue. Do you still have this problem with the current version? I fixed a whole bunch of permission problems after first publishing this, which unfortunately were not pushed out to the DockerHub image. Later on, I created an autobuild version.

With the current version, I am not seeing this error:

```
# mkdir -p /docker/DATA/misp/mysql
# docker run -it --rm \
-v /docker/DATA/misp/mysql:/var/lib/mysql  \
harvarditsecurity/misp /init-db

# docker run -it -d \
--name misp \
-p 443:443 \
-p 80:80 \
-p 3306:3306 \
-v /docker/data/misp/mysql:/var/lib/mysql \
harvarditsecurity/misp

d69ae74dc239c12fe3eb2e1543485e369742bbada4e79230803c4c1a5f0cdd5c

# docker ps
CONTAINER ID        IMAGE                    COMMAND                  CREATED             STATUS              PORTS                                                              NAMES
d69ae74dc239        harvarditsecurity/misp   "/usr/bin/supervisord"   9 seconds ago       Up 7 seconds        0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp, 0.0.0.0:3306->3306/tcp   misp
```

^ I can login without any problems

If you have a few minutes, could you please try it from your side?

Thanks!

**Comment by radder5** [comment]
Hi,
I have a similar issue (looks the same at first) i.e. can't install system tables.

```
Installing MariaDB/MySQL system tables in '/var/lib/mysql' ...
170209 12:33:48 [Note] /usr/sbin/mysqld (mysqld 10.0.29-MariaDB-0ubuntu0.16.04.1) starting as process 39 ...

Installation of system tables failed!  Examine the logs in
/var/lib/mysql for more information.

The problem could be conflicting information in an external
my.cnf files. You can ignore these by doing:

    shell> /usr/scripts/scripts/mysql_install_db --defaults-file=~/.my.cnf

You can also try to start the mysqld daemon with:

    shell> /usr/sbin/mysqld --skip-grant --general-log &

and use the command line tool /usr/bin/mysql
to connect to the mysql database and look at the grant tables:

    shell> /usr/bin/mysql -u root mysql
    mysql> show tables;

Try 'mysqld --help' if you have problems with paths.  Using
--general-log gives you a log in /var/lib/mysql that may be helpful.

The latest information about mysql_install_db is available at
https://mariadb.com/kb/en/installing-system-tables-mysql_install_db
MariaDB is hosted on launchpad; You can find the latest source and
email lists at http://launchpad.net/maria

Please check all of the above before submitting a bug report
at http://mariadb.org/jira

170209 12:33:49 mysqld_safe Logging to syslog.
170209 12:33:49 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
```

I get no errors reported in log files in /Users/conradc/misp-db (/var/lib/mysql) but I get a bunch of files appear in that directory - aria_log_control, aria_log.0000001, ib_logfile1, etc.
None of which are txt files.

Running on Mac (via Kitematic)

Any suggestions?

Thanks
Conrad

**Comment by ventz** [comment]
@radder5 Could you try the launching it via the steps manually just to eliminate an issue with Kitematic?

I've tried this on a couple of different systems (new ones, without having it previously setup) and it seems to work.

**Comment by radder5** [comment]
I have tried this on a Linux server with Docker and still have issues with the permissions of the mysql  data directory without setting the permissions up directly as in the from robinlennox above. But after running this 

`docker run -it --rm     -v /misp-db:/var/lib/mysql harvarditsecurity/misp /bin/bash -c "/etc/rc.local; chown -R mysql:mysql /var/lib/mysql"`

it works (in principle). I have had to add some other elements to the Dockerfile in order to get the pgp keys generated as it was erroring with 'not enough entropy - do some more work', but the build was getting hung at this point. I added in rng-tools and some other lines to ensure this would work and got through this problem.

So, in summary I have this running ok now on a Linux version of Docker. As I am intending to add support for modules too into this Dockerfile I'll see if I can include the permissions change to the mysql data directory too (but I thought I had seen that already).

Are you interested in getting my revised Dockerfile if I get the module installation working ok?

Thanks
Conrad

**Comment by ventz** [comment]
@radder5 Interesting.

I've already added the chown to the next Dockerfile release already, so I can push it out + build a new version sooner just to get it out.

I wonder if because the directory is in /, it somehow gets impacted permissions wise when it gets mounted?

About having enough random entropy - the concern is that as soon as you add rng-tools, you are now getting pseudo-random numbers. For most people, this will be fine, but at this point it's not technically secure. I am ok with adding it as long as we can properly document this and make the end-user aware.

Always open to improvements and merges by the way -- if you add something useful/etc, submit a pull request and we can merge it in. My goal was to 1.) create an easy way to deploy this while still adding a way to set all of the options/bells and whistles, and 2.) making it as useful as possible towards the community.

**Comment by ventz** [comment]
@radder5 Pushed out the new version to github, including the chown I added the rng-tools + a note about it. Building a new image now and will push it out to dockerhub. Thanks for taking the time to test it out and for the feedback. Again, as you add improvements, feel free to submit pull requests and we can merge them.

**Comment by ventz** [comment]
Just closing this off with a summary for anyone that has already downloaded the image.

This issue was fixed in commit: https://github.com/harvard-itsecurity/docker-misp/commit/09c043f779f8a2e2618112cce2b46f0d6faf52cb

Thank you everyone for the feedback!
